core/state_transition.go

```diff
@@ -282,6 +280,18 @@ func (st *StateTransition) refundGas() {
 	st.gp.AddGas(st.gas)
 }
 
+func (st *StateTransition) collectGas() {
+	// Burn Txn Fees after staking epoch
+	if config := st.evm.ChainConfig(); !config.IsStaking(st.evm.EpochNumber) {
+		txFee := new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.gasPrice)
+		st.state.AddBalance(st.evm.Coinbase, txFee)
+	} else if config.IsFeeCollectEpoch(st.evm.EpochNumber) { // collect Txn Fees to community-managed account.
```

Comment:
should this be the first condition that is checked? otherwise after this fork, `IsStaking` will always be true and you may never hit `IsFeeCollectEpoch` condition? I think the `IsFeeCollectEpoch` condition should come first and fallback `else` should check `IsStaking` for burning?

Reply:
The first condition is "!config.IsStaking(st.evm.EpochNumber)"  so right now, it's always false. The catch is fee collection only works after staking epoch. If it's set before staking epoch, the original logic of fee going to the miner take precedence.

@peekpi please add the catch as a comment to the FeeCollectionEpoch definition.

Reply:
> ...
> @peekpi please add the catch as a comment to the FeeCollectionEpoch definition.

Also add function `mustValid()` to internal/params/config.go to ensure `ChainConfig` is valid.